packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -772,9 +851,12 @@ export default class GoogleCalendarService implements Calendar {
     }
     const getCalIds = async () => {
       if (selectedCalendarIds.length !== 0) return selectedCalendarIds;
-      const cals = await this.getAllCalendars(calendar, ["id"]);
+      const cals = await this.getAllCalendars(calendar, ["id", "primary"]);
       if (!cals.length) return [];
-      return cals.reduce((c, cal) => (cal.id ? [...c, cal.id] : c), [] as string[]);
+      if (!fallbackToPrimary) return this.getValidCalendars(cals).map((cal) => cal.id);
+      const primaryCalendar = this.filterPrimaryCalendar(cals);
+      if (!primaryCalendar) return [];
+      return [primaryCalendar.id];
```

Comment:
This is the logic that ensures that primary calendar will always be checked for conflict, even if the user hasn;'t toggle on any calendar for conflict checking